### PR TITLE
5th PR - Implemented Fragments

### PR DESCRIPTION
**Pull request by AndresConfortiGL**
## Description
5th PR - A dialog fragment was implemented in order to see the character information.

## Story Type
- New feature :white_check_mark:
- Change request
- Bug fix
- Tech

## How Can This Be Tested?
By pressing a character, you will be prompted by a fragment with the character information.

## Evidences
| Pixel 3a | Nexus S |
| ------------- | ------------- |
|pixel|nexus|

| Coverage Test  |
| ------------- |
|coverage|

## Checklist
- All unit Test passed :white_check_mark:

## Tested on
The project was tested on Android 8.0 / API 26 using the following devices:

- Pixel 3a
- Nexus S

## Package Structure
| Structure |
| ------------- |
|package|

### CODE REVIEW

File: app/src/main/java/com/example/marvelapp/service/api/MarvelApi.kt [hunk 1]
```diff
@@ -1,10 +1,16 @@
...
 import com.example.marvelapp.service.response.DataResponse
 import retrofit2.Call
 import retrofit2.http.GET
+import retrofit2.http.Path
 
 interface MarvelApi {
     @GET("/v1/public/characters")
     fun getCharacters(): Call<DataResponse>
+
+    //spider ham id - 1011347
```
Comment (SartiMau) [inline app/src/main/java/com/example/marvelapp/service/api/MarvelApi.kt]: please, remove this comment

File: app/src/main/res/layout/character_fragment.xml [hunk 2]
```diff
@@ -0,0 +1,48 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    android:layout_margin="25dp"
```
Comment (SartiMau) [inline app/src/main/res/layout/character_fragment.xml]: do not hardcode, use dimens please

**Review by SartiMau — CHANGES_REQUESTED** [review]

**Review by Georgina-Fernandez — APPROVED** [review]

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -4,6 +4,9 @@
     <string name="main_activity_description">PRESS THE BUTTON TO GET A LIST WITH 20 MARVEL CHARACTERS ALPHABETICALLY ORDERED.</string>
     <string name="main_activity_button">PRESS ME</string>
     <string name="characters_activity_get_error">ERROR - CANT FETCH DATA</string>
+    <string name="fragment_no_description">NO DESCRIPTION</string>
+    <string name="fragment_name">NAME</string>
     <string name="characters_activity_title">CHARACTERS LIST</string>
     <string name="cardview_image_description">DESCRIPTION</string>
+    <string name="loren">Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse suscipit risus neque, consequat suscipit velit semper ut. Mauris blandit pretium nulla, sed egestas ex FIN.</string>
```
Comment (DiegoDiazGlobant) [inline app/src/main/res/values/strings.xml]: Do you use this string? `loren`

File: app/src/main/java/com/example/marvelapp/fragment/CharacterFragment.kt [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+class CharacterFragment : DialogFragment(), KoinComponent {
+    private lateinit var binding: CharacterFragmentBinding
+    private val viewModel: FragmentViewModel by inject()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = CharacterFragmentBinding.inflate(layoutInflater)
+        viewModel.fragmentState.observe({ lifecycle }, ::fragmentState)
+        viewModel.getCharacter(arguments?.getString(Constants.CHARACTER_ID).orEmpty())
+        dialog?.window?.setBackgroundDrawableResource(android.R.color.transparent)
```
Comment (DiegoDiazGlobant) [inline app/src/main/java/com/example/marvelapp/fragment/CharacterFragment.kt:26]: nice!

File: app/src/main/java/com/example/marvelapp/service/api/MarvelApi.kt (hunk 1 above)
Comment (AndresConfortiGL) [reply app/src/main/java/com/example/marvelapp/service/api/MarvelApi.kt]: fixed

File: app/src/main/res/layout/character_fragment.xml (hunk 2 above)
Comment (AndresConfortiGL) [reply app/src/main/res/layout/character_fragment.xml]: fixed

File: app/src/main/res/values/strings.xml (hunk 3 above)
Comment (AndresConfortiGL) [reply app/src/main/res/values/strings.xml]: Fixed. I forgot to remove it.

File: app/src/main/java/com/example/marvelapp/fragment/CharacterFragment.kt (hunk 4 above)
Comment (AndresConfortiGL) [reply app/src/main/java/com/example/marvelapp/fragment/CharacterFragment.kt:26]: Thanks!

**Review by DiegoDiazGlobant — APPROVED** [review]

File: app/src/main/res/layout/character_fragment.xml [hunk 5]
```diff
@@ -0,0 +1,48 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    android:layout_margin="@dimen/cardview_image_border"
+    android:minWidth="300dp"
```
Comment (SartiMau) [inline app/src/main/res/layout/character_fragment.xml]: do not hardcode

**Review by SartiMau — CHANGES_REQUESTED** [review]

**Review by SartiMau — APPROVED** [review]

**Pull request merged**